com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/visitor/CobolVisitor.java

```diff
@@ -38,24 +41,140 @@
   private static final int INFO_LEVEL = 3;
 
   @Getter private List<SyntaxError> errors = new ArrayList<>();
+  @Getter @Setter private String programName;
 
   private String documentUri;
   private SemanticContext semanticContext;
+  private CommonTokenStream tokenStream;
 
-  public CobolVisitor(String documentUri, SemanticContext semanticContext) {
+  public CobolVisitor(
+      String documentUri, SemanticContext semanticContext, CommonTokenStream tokenStream) {
     this.documentUri = documentUri;
     this.semanticContext = semanticContext;
+    this.tokenStream = tokenStream;
+  }
+
+  @Override
+  public Class visitIdentificationDivision(IdentificationDivisionContext ctx) {
+    String token = ctx.getStart().getText();
+    areaAWarning(ctx.start.getCharPositionInLine(), token, ctx.getStart().getLine());
```

Comment:
Maybe, it would be better to extract `ctx.getStart()` as a variable to reduce duplicated calls?

Reply:
sure

Reply:
👍 I left a similar comment, I will resolve mine :)